File: src/main/java/me/viniciusarnhold/altaria/events/interfaces/ICommandHandler.java
package me.viniciusarnhold.altaria.events.interfaces;

import me.viniciusarnhold.altaria.events.utils.Commands;
import org.jetbrains.annotations.Nullable;
import sx.blah.discord.handle.impl.events.MessageReceivedEvent;
import sx.blah.discord.util.DiscordException;
import sx.blah.discord.util.MissingPermissionsException;
import sx.blah.discord.util.RateLimitException;

import java.util.List;

/**
 * Created by Vinicius.
 *
 * @since 1.0
 */
public interface ICommandHandler {

    boolean handle(MessageReceivedEvent event, String command, String matchedText) throws RateLimitException, DiscordException, MissingPermissionsException;

    @Nullable
    List<Commands> getHandableCommands();

}

File: core/src/main/kotlin/me/viniciusarnhold/altaria/events/interfaces/ICommandHandler.kt
package me.viniciusarnhold.altaria.events.interfaces

import me.viniciusarnhold.altaria.events.utils.Commands
import sx.blah.discord.handle.impl.events.MessageReceivedEvent
import sx.blah.discord.util.DiscordException
import sx.blah.discord.util.MissingPermissionsException
import sx.blah.discord.util.RateLimitException

/**
 * Created by Vinicius.

 * @since 1.0
 */
interface ICommandHandler {

    @Throws(RateLimitException::class, DiscordException::class, MissingPermissionsException::class)
    fun handle(event: MessageReceivedEvent, command: String, matchedText: String): Boolean

    val handableCommands: List<Commands>?

}
